test/test_common/environment.cc

```diff
@@ -129,6 +179,20 @@ std::string TestEnvironment::getCheckedEnvVar(const std::string& var) {
 }
 
 void TestEnvironment::initializeOptions(int argc, char** argv) {
+  // Before latching argv and argc, remove any runtime override flag.
+  // This allows doing test overrides of Envoy runtime features without adding
+  // test flags to the Envoy production command line.
+  const std::regex PATTERN{"--runtime-feature-override-for-tests=(.*)", std::regex::optimize};
```

Comment:
I have a probably stupid question: Is there any reason we couldn't just hit this directly from a test given that we already have static support and also have some work that @jmarantz did to clean things up at the end of tests? I'm just wondering if we could avoid some of the regex/listener/etc. magic here.

Reply:
Hm, now that I did the const-singleton refactor this is much less tied to environment.cc than it was.

I'd be inclined to forklift all this code over to TestRunner::RunTests - we still need the regex magic and I'd prefer a separate listener just to avoid the set up and teardown checks for the vast majority of tests which don't need them.

I'm going to hold off until we've decided what to do with command line flags - if we want to make it general purpose I'd be inclined to leave it here with the TODO, and the move it into the main server TCLAP in the follow-up.

Reply:
I think my feeling is to not making it general purpose for now, and just track somewhere in a TODO/issue/etc. that we might want to think about this in the future. So in that case, I guess I would opt for making the test code as simple as possible so maybe do the move you suggested? Though if we do a move, why do we need all the regex stuff? Couldn't we just hit a static method with a string to alter runtime state for a test?

Reply:
Oh, we can do this statically from test, but in-house when introducing a high risk flag we often just dup a full test (or tests) with no edits, so we have one variant running with true and the other running with the flag false.  We've found it pretty handy especially for tests which are already parameterized so hard to muck with.

It's not a lot of code so I'd be inclined to check it in and we can always remove it if no one uses it and it turns out to be overkill. WDYT?

Reply:
I don't have a strong opinion on this one way or the other, but couldn't we accomplish roughly the same thing with a TEST_P and some type of param helper that just toggles a specific string? I think @dnoe mentioned this? I'm mainly just wondering if the same behavior can be achieved with less code?

Reply:
We can, but if a test is already a TEST_P it can mean updating all the test params from GetParam() to GetParam(0), adding GetParam(1) for every test fixture in the whole class, creating or editing the constructur etc, then tearing it all back down in a few months.  Given many of our files have 3-5 fixtures this can end up being a lot of throwaway work per test you want to dual-dun, which is why we bother with the flag based solution internally.

Reply:
OK SGTM if you think this is the best option.

Reply:
Again it's really handy to have, but TDB if folks use it.
I've added a calendar reminder 6 months out to revisit - if no one uses it I'll tear these changes out then.

Any other comments you're waiting on?